spec/betterc.dd

```diff
@@ -0,0 +1,48 @@
+Ddoc
+
+$(SPEC_S Better C,
+
+`-betterC` is a Command-Line flag for `dmd`, which generated light-weight,
+barebone, runtime-free binaries with are suitable for linking from C.
+
+---
+void main()
+{
+    import core.stdc.stdio : printf;
+    printf("Hello betterC");
+}
+---
+
+$(CONSOLE
+> dmd -betterC hello.d
+Hello betterC
+)
+
+$(H2 $(LNAME2 effects, Effects))
+
+    $(OL
+        $(LI Doesn't link DRuntime and standard library by default)
+	    $(LI Doesn't generate ModuleInfo entries)
+    )
+
+$(H2 $(LNAME2 consequences, Consequences))
+
+As no DRuntime is available, many D features won't work. For example:
+
+	$(OL
+	    $(LI No static module constructors or deconstructors)
+	    $(LI No Garbage Collection)
+	    $(LI No dynamic arrays and associative arrays)
+	    $(LI No Classes)
+	    $(LI No `synchronized`, monitors, mutex from $(MREF core, sync))
```

Comment:
That's a bummer, though since you can't define a parameter-less constructor on a `struct`s, you can't mimic C++11's `std::mutex` anyway. I'm leaning towards defining empty (default) ctor and dtor and two methods `create()` and `release()` , which is obviously not very safe, but it's meant to be a low-level implementation wrapped by a higher-level one like `core.sync.mutex.Mutex` or a factory of some sort (with good enough API reusing mutexes can be both safer and faster).